universe/UnlockableItem.h

```diff
@@ -0,0 +1,61 @@
+#ifndef _UnlockableItem_h_
+#define _UnlockableItem_h_
+
+
+#include <GG/Enum.h>
+#include "../util/Export.h"
+
+
+/** types of items that can be unlocked for empires */
+GG_ENUM(UnlockableItemType,
+    INVALID_UNLOCKABLE_ITEM_TYPE = -1,
+    UIT_BUILDING,               ///< a kind of Building
+    UIT_SHIP_PART,              ///< a kind of ship part (which are placed into hulls to make designs)
+    UIT_SHIP_HULL,              ///< a ship hull (into which parts are placed)
+    UIT_SHIP_DESIGN,            ///< a complete ship design
+    UIT_TECH,                   ///< a technology
+    NUM_UNLOCKABLE_ITEM_TYPES   ///< keep last, the number of types of unlockable item
+)
+
+
+//! Specifies single item of game content that may be unlocked for an empire.
+//! The @a type field stores the type of item that is being unlocked, such as
+//! building or ship component, and the  @a name field contains the name of the
+//! actual item (e.g. (UIT_BUILDING, "Superfarm") or
+//! (UIT_SHIP_PART, "Death Ray")).
+struct FO_COMMON_API UnlockableItem {
+    UnlockableItem();
+    UnlockableItem(UnlockableItemType type_, const std::string& name_) :
+        type(type_),
+        name(name_)
+    {}
+
+    //! Returns a data file format representation of this object
+    auto Dump(unsigned short ntabs = 0) const -> std::string;
```

Comment:
What's the purpose of using `auto` and `-> std::string` here?

Reply:
It's for consistent indentation.  `void` and `auto` have the same width and by using trailing return types there is no need for whitespace fuckery (or "grooming").